### ISSUE TITLE
`#include "tsl/framework/numeric_types.h"` is missing

### ISSUE THREAD

**Issue by JanuszL** [tensorflow/tensorflow#61830]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

source

### TensorFlow version

2.15

### Custom code

Yes

### OS platform and distribution

docker/tensorflow/tensorflow:nightly-gpu

### Mobile device

no

### Python version

3.11

### Bazel version

NR

### GCC/compiler version

NR

### CUDA/cuDNN version

NR

### GPU model and memory

NR

### Current behavior?

Create test.cc
```
#include "tensorflow/core/framework/op.h"
```
compile it with flags provided by TensorFlow:
get_compile_flags():
```['-I/usr/local/lib/python3.11/dist-packages/tensorflow/include', '-D_GLIBCXX_USE_CXX11_ABI=1', '--std=c++17', '-DEIGEN_MAX_ALIGN_BYTES=64']```
get_link_flags():
```['-L/usr/local/lib/python3.11/dist-packages/tensorflow', '-l:libtensorflow_framework.so.2']```
Then
```
g++ test.cc -o /tmp/test.o -fPIC -I/usr/local/lib/python3.11/dist-packages/tensorflow/include -D_GLIBCXX_USE_CXX11_ABI=1 --std=c++17 -DEIGEN_MAX_ALIGN_BYTES=64 -L/usr/local/lib/python3.11/dist-packages/tensorflow -l:libtensorflow_framework.so.2  -O2```
The error
```
In file included from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/bfloat16.h:19,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/types.h:24,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op_def_builder.h:28,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/full_type_inference_util.h:24,
                 from /usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/op.h:27,
                 from test.cc:1:
/usr/local/lib/python3.11/dist-packages/tensorflow/include/tensorflow/core/framework/numeric_types.h:24:10: fatal error: tsl/framework/numeric_types.h: No such file or directory
   24 | #include "tsl/framework/numeric_types.h"
```

### Standalone code to reproduce the issue

```shell
As above
```

### Relevant log output

```shell
As above
```

**Comment by sachinprasadhs** [comment]
This could be due to the commit [Vendor XLA/TSL into TensorFlow as Bazel dependencies](https://github.com/tensorflow/tensorflow/commit/132c98b29c1a42298cd13055d1168c8d67177bc9)
And the below files are affected for `numberic_types.h `
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/framework/numeric_types.h
https://github.com/tensorflow/tensorflow/blob/master/third_party/xla/third_party/tsl/tsl/framework/numeric_types.h

@jakeharmon8 , Could you please take a look into this. Thanks!

**Comment by JanuszL** [comment]
Hi @sachinprasadhs,

It seems to work now.

**Comment by sachinprasadhs** [comment]
Thanks for confirming, could you pease close this issue.

**Comment by njzjz** [comment]
I still got this error in 2.15.0rc0 on Windows. The package was installed using `pip`.

```
    C:/Users/runneradmin/AppData/Local/Temp/pip-build-env-csrgykur/normal/Lib/site-packages/tensorflow/include\tensorflow/core/framework/numeric_types.h(24,1): fatal error C1083: Cannot open include file: 'tsl/framework/numeric_types.h': No such file or directory [D:\a\deepmd-kit\deepmd-kit\build\py37-none-win_amd64\op\deepmd_op.vcxproj]
      Building Custom Rule D:/a/deepmd-kit/deepmd-kit/source/op/CMakeLists.txt
      cmake_pch.cxx
    C:/Users/runneradmin/AppData/Local/Temp/pip-build-env-csrgykur/normal/Lib/site-packages/tensorflow/include\tensorflow/core/framework/numeric_types.h(24,1): fatal error C1083: Cannot open include file: 'tsl/framework/numeric_types.h': No such file or directory [D:\a\deepmd-kit\deepmd-kit\build\py37-none-win_amd64\op\op_grads.vcxproj]
```

There are no errors on Linux and macOS, though.

**Comment by njzjz** [comment]
> There are no errors on Linux and macOS, though.

I checked the Linux wheel and the Windows wheel. The Linux wheel contains `include/tsl` and `include/xla` directories, but the Windows wheel doesn't.

**Comment by JanuszL** [comment]
This time I see:
```
fatal error: third_party/eigen3/unsupported/Eigen/CXX11/Tensor: No such file or directory
         23 | #include "third_party/eigen3/unsupported/Eigen/CXX11/Tensor"```

**Comment by njzjz** [comment]
> I checked the Linux wheel and the Windows wheel. The Linux wheel contains `include/tsl` and `include/xla` directories, but the Windows wheel doesn't.

Hi @sachinprasadhs, this issue still exists in 2.15.0rc1. I know now Intel builds the Windows package. Is there some way to report to them?

**Comment by sachinprasadhs** [comment]
cc: @mraunak

**Comment by mraunak** [comment]
Hi @njzjz and @sachinprasadhs, the above issue will be fixed in the next release TF 2.16

**Comment by njzjz** [comment]
> the above issue will be fixed and missing xla/tsl files will be available in the include folder in the next release TF 2.16.

I confirm it has been fixed in TF 2.16.

**Comment by mihaimaruseac** [comment]
Closing since it is fixed. Thank you for the confirmation
